### PR TITLE
update remove and add book

### PR DESCRIPTION
**Pull request by Bigizi**
**Store Book React Project**

I connected the configureStore file to the main component of your React app.
I added a slice that contains my books
I displayed books, received from the slice, in a reusable component
I add an event handler to add and remove buttons

### CODE REVIEW

File: src/components/Book.js [hunk 1]
```diff
@@ -1,14 +1,22 @@
...
-import PropTypes from 'prop-types';
+// import PropTypes from 'prop-types';
+import { useDispatch, useSelector } from 'react-redux';
+import { removeBook } from '../redux/books/Books';
 
-const Book = (props) => {
-  const { book } = props;
+const Book = () => {
+  // const { book } = props;
+  const book = useSelector((state) => state.books);
```
Comment (anagudelogu) [inline src/components/Book.js]: - [ ] Please don't forget to remove old pieces of commented code, they make your code base look ugly and make it less readable. 🤓 🔧

File: src/components/Book.js [hunk 2]
```diff
@@ -21,22 +29,16 @@ const Book = (props) => {
           <p className="para" style={styles.para}>{book.title}</p>
           <p>{book.author}</p>
           <span>
-            <button type="button">Remove</button>
+            {/* <button type="button">Comments</button> */}
+            <button type="button" onClick={() => handleRemove(book.id)}>
+              Remove
+            </button>
+            {/* <button type="button">Edit</button> */}
```
Comment (anagudelogu) [inline src/components/Book.js]: - [ ] Please don't forget to remove old pieces of commented code, they make your code base look ugly and make it less readable. 🤓 🔧

File: src/components/Book.js [hunk 3]
```diff
@@ -1,14 +1,22 @@
...
-const Book = (props) => {
-  const { book } = props;
+const Book = () => {
+  // const { book } = props;
+  const book = useSelector((state) => state.books);
+  const dispatch = useDispatch();
+
+  const handleRemove = (bookId) => {
+    dispatch(removeBook(bookId));
+    console.log(bookId);
```
Comment (anagudelogu) [inline src/components/Book.js]: - [ ] According to the [JavaScript best practices](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/javascript_best_practices.md):

image

Please consider removing this `console.log()` statement! 🤓 🔧

File: src/components/routes/Books.js [hunk 4]
```diff
@@ -1,21 +1,15 @@
 import React from 'react';
-import { v4 as uuidv4 } from 'uuid';
+// import { v4 as uuidv4 } from 'uuid';
```
Comment (anagudelogu) [inline src/components/routes/Books.js]: - [ ] Please don't forget to remove old pieces of commented code, they make your code base look ugly and make it less readable. 🤓 🔧

File: src/components/routes/Categories.js [hunk 5]
```diff
@@ -1,13 +1,19 @@
...
+import { checkStatus } from '../../redux/categories/Categories';
 
 const Categories = () => {
-  const handleClick = () => {
-    // console.log('Under construction');
-  };
+  const dispatch = useDispatch();
+  // const handleClick = () => {
+  //   // console.log('Under construction');
+  // };
```
Comment (anagudelogu) [inline src/components/routes/Categories.js]: - [ ] Please don't forget to remove old pieces of commented code, they make your code base look ugly and make it less readable. 🤓 🔧

**Review by anagudelogu — CHANGES_REQUESTED** [review]
# STATUS:  CHANGES REQUIRED  ♻️ 🚧 

Hi @Bigizi, 
I'm Andres.


While you have done a great job working on this milestone 💻 , there are some issues that you still need to work on to go to the next project. ♻️ 

You are almost there, keep up the good work! 🚀 



## Highlights
✅ Great use of Git flow. 💯 
✅ Your project is documented professionally. 📜 
✅ Great UI design. 💻 
✅ Descriptive PR title, summary and commit messages. 💬 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Ghiftee — APPROVED** [review]
## STATUS: APPROVED ♻️♻️
Hi @Bigizi, 

To highlight 👍🏽 
✔️ README is professional
✔️ Codebase is clean
✔️ No errors

Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀  
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**Pull request merged**